subsys/net/lib/nrf_cloud/src/nrf_cloud.c

```diff
@@ -535,7 +535,8 @@ void nrf_cloud_run(void)
 	atomic_set(&transport_disconnected, 0);
 
 	while (true) {
-		ret = poll(fds, ARRAY_SIZE(fds), POLL_TIMEOUT_MS);
+		ret = poll(fds, ARRAY_SIZE(fds),
+			cloud_keepalive_time_left(nrf_cloud_backend));
 
 		if (ret == 0) {
 			if (cloud_keepalive_time_left(nrf_cloud_backend) <
```

Comment:
AFAIK poll returns 0 when the timeout expires. Then this check is redundant and the define ```POLL_TIMEOUT_MS``` can be removed.